apps/network-alerts/test/unit/attention/notify_test.js

```diff
@@ -0,0 +1,244 @@
+    var o1 = AudioContext.prototype.createOscillator.firstCall.returnValue;
+    var o2 = AudioContext.prototype.createOscillator.secondCall.returnValue;
+
+    assert.equal(o1.type, 'sine');
+    assert.equal(o2.type, 'sine');
+
+    // either one case or the other
+    try {
+      assert.equal(o1.frequency.value, 853);
+      assert.equal(o2.frequency.value, 960);
+    } catch(e) {
+      assert.equal(o2.frequency.value, 853);
+      assert.equal(o1.frequency.value, 960);
+    }
+
+    sinon.assert.calledOn(o1.connect, o1);
+    sinon.assert.calledOn(o2.connect, o2);
+    sinon.assert.alwaysCalledWith(o1.connect, gainNode);
+    sinon.assert.calledWith(gainNode.connect, audioCtx.destination);
+
+    sinon.assert.calledOn(o1.start, o1);
+    sinon.assert.calledOn(o2.start, o2);
+    sinon.assert.calledOn(o1.stop, o1);
+    sinon.assert.calledOn(o2.stop, o2);
+    sinon.assert.alwaysCalledWith(o1.stop, 11);
+
+    sinon.assert.calledOn(gainNode.gain.setValueCurveAtTime, gainNode.gain);
+    sinon.assert.calledWith(
+      gainNode.gain.setValueCurveAtTime,
+      sinon.match.instanceOf(Float32Array), 0, 11
+    );
+  }
+
+  function setSettings(settings) {
+    for (var key in settings) {
+      getStub.withArgs(key).returns(Promise.resolve({
+        [key]: settings[key]
+      }));
+    }
+  }
+
+  suite('mozSettings is unavailable', function() {
+    test('We do nothing but we do not stop', function(done) {
+      delete navigator.mozSettings;
+
+      Notify.notify().then(() => {
+        sinon.assert.notCalled(AudioContext);
+        sinon.assert.notCalled(navigator.vibrate);
+      }).then(done, done);
+    });
+  });
+
+  suite('volume is 1 and vibration is enabled', function() {
+    test('play ringtone and vibrate', function(done) {
+      setSettings({
+        'audio.volume.notification': 1,
+        'vibration.enabled': true
+      });
+
+      Notify.notify().then(() => {
+        // As document is not shown in the test, we launch the
+        // event of visibility
+        sendVisibilityEvent();
+        assertAudio();
+        assertVibrate();
+      }).then(done, done);
+    });
+  });
+
+  suite('volume is 1 and vibration is disabled', function() {
+    test('play ringtone and do not vibrate', function(done) {
+      setSettings({
+        'audio.volume.notification': 1,
+        'vibration.enabled': false
+      });
+
+      Notify.notify().then(() => {
+        sinon.assert.notCalled(navigator.vibrate);
+        assertAudio();
+      }).then(done, done);
+    });
+  });
+
+  suite('volume is 0 and vibration is enabled', function() {
+    test('do not play ringtone and vibrate', function(done) {
+      setSettings({
+        'audio.volume.notification': 0,
+        'vibration.enabled': true
+      });
+
+      Notify.notify().then(() => {
+        // As document is not shown in the test, we launch the
+        // event of visibility
+        sendVisibilityEvent();
```

Comment:
shouldn't we verify that vibrate hasn't been called before visibility change event?